### PR TITLE
Introduce Server V1 Notifier facade

### PR DESCRIPTION
**Pull request by azdagron**
- Adds v1 to the catalog and deprecates v0
- Migrates the gcs_bundle notifier to v1

### CODE REVIEW

File: pkg/server/plugin/notifier/v1.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+type V1 struct {
+	plugin.Facade
+	notifierv1.NotifierPluginClient
+}
+
+func (v1 *V1) NotifyAndAdviseBundleLoaded(ctx context.Context, bundle *common.Bundle) error {
+	pluginBundle, err := bundleFromCommonProto(bundle)
+	if err != nil {
+		return v1.Errorf(codes.InvalidArgument, "bundle is malformed: %v", err)
```
Comment (amartinezfayo) [inline pkg/server/plugin/notifier/v1.go:22]: This path is not exercised by the tests, we may add a case relatively easily I think.

File: pkg/server/plugin/notifier/v1.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+			},
+		},
+	})
+	return v1.WrapErr(err)
+}
+
+func (v1 *V1) NotifyBundleUpdated(ctx context.Context, bundle *common.Bundle) error {
+	pluginBundle, err := bundleFromCommonProto(bundle)
+	if err != nil {
+		return v1.Errorf(codes.InvalidArgument, "bundle is malformed: %v", err)
```
Comment (amartinezfayo) [inline pkg/server/plugin/notifier/v1.go:37]: This path is not exercised by the tests, we may add a case relatively easily I think.

**Review by amartinezfayo — COMMENTED** [review]
Looking great!

File: pkg/server/plugin/notifier/v1.go (hunk 1 above)
Comment (azdagron) [reply pkg/server/plugin/notifier/v1.go]: SGTM!

File: pkg/server/plugin/notifier/v1.go (hunk 2 above)
Comment (azdagron) [reply pkg/server/plugin/notifier/v1.go]: SGTM!

**Review by amartinezfayo — APPROVED** [review]
🚀

**Pull request merged**